OpenSourcePolitics/decidim-module_homepage_proposals
# Fix/glide 0.26

#### Description 

Fix bugs and enhance content block 


#### Done

- [x] Disable scroll when no proposals found
- [x] Ensure active glide__bullets reflects the current card in carousel
- [x] Tweak breakpoints when proposals length is between 1 and 3
- [x] Add 'All proposals' button based on current component ID
- [x] Transform Desktop filters to in row mixin with translation keys
- [x] Add missing i18n specs
- [x] **Bump to v1.0.1** 

## How to test

Run the module locally 

* Clone module
* Use node v16 (check with `node -v`)
* Install ruby and JS dependencies `bundle install && yarn install`
* Generate a new dev application : `bundle exec rake development_app`
* Move in newly created application : `cd development_app`
* Compile assets with webpacker : `bundle exec bin/webpack`
* Run server : `bundle exec rails s`

Then : 

* Configure content block in backoffice
* Ensure slider is functional in homepage 
* If more than 4 proposals found : 
- User can slide using bullets, keyboard arrow and mouse slide
- It should not slide automatically
* If less than 4 proposals:
- Carousel should be disabled

When using filters :
* Slider items should be updated asynchronously 

When resizing window : 
* Items presented should be reduced from 4, to 3, to 2 to 1 related to the current window size

Try to find edge cases with private components, created proposals, etc...

### 🚨 known issues

* In development mode, if you filter at least once and then resize the window a JS error happens, it is not reproducible in production mode
